src/generic/canbus.c

```diff
@@ -153,6 +161,21 @@ can_process_set_nodeid(uint32_t id, uint32_t len, uint8_t *data)
     }
 }
 
+static void
+can_process_request_bootloader(uint32_t id, uint32_t len, uint8_t *data)
+{
+    if (!can_check_uuid(id, len, data))
+        return;
+    irq_disable();
+    uint32_t *bl_vectors = (uint32_t *)(CONFIG_FLASH_START & 0xFF000000);
+    uint64_t *boot_sig = (uint64_t *)(bl_vectors[1] - 9);
+    if (*boot_sig == CANBOOT_SIGNATURE) {
```

Comment:
We also need to check for alignment here - as some ARM platforms will crash on an unaligned integer access.  Something like:
```c
    if (boot_sig == ALIGN(boot_sig, 8) && boot_sig == CANBOOT_SIGNATURE && req_sig == ALIGN(req_sig, 8)) {
        ...
```

I guess the ResetHandler assembler should also have a `.balign 8` directive just before the CANBOOT_SIGNATURE to ensure it is 8-byte aligned.

Reply:
This should be fixed .  I had to do some type casting to keep the compiler from complaining, but the result should be correct.  I also added `balign 8` to the asm in CanBoot.